src/rules/noDuplicateImportsRule.ts

```diff
@@ -44,13 +44,38 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 }
 
-function walk(ctx: Lint.WalkContext<void>) {
-    const seen = new Set<string>();
-    for (const {text, parent} of findImports(ctx.sourceFile, ImportKind.ImportDeclaration)) {
-        if (seen.has(text)) {
-            ctx.addFailureAtNode(parent!, Rule.FAILURE_STRING(text));
-        } else {
+function walk(ctx: Lint.WalkContext<void>): void {
+    walkWorker(ctx, ctx.sourceFile.statements, new Set());
+}
+
+function walkWorker(ctx: Lint.WalkContext<void>, statements: ReadonlyArray<ts.Statement>, seen: Set<string>): void {
+    for (const statement of statements) {
+        const im = tryGetImportSpecifier(statement);
+        if (im !== undefined && isLiteralExpression(im)) {
+            const { text } = im;
+            if (seen.has(text)) {
+                ctx.addFailureAtNode(im.parent!, Rule.FAILURE_STRING(text));
+            }
             seen.add(text);
         }
+
+        if (isModuleDeclaration(statement) && statement.body !== undefined && isModuleBlock(statement.body)) {
+            // If this is a module augmentation, re-use `seen` since those imports could be moved outside.
+            // If this is an ambient module, create a fresh `seen`
+            // because they should have separate imports to avoid becoming augmentations.
+            walkWorker(ctx, statement.body.statements, ts.isExternalModule(ctx.sourceFile) ? seen : new Set());
+        }
+    }
+}
+
+function tryGetImportSpecifier(statement: ts.Statement): ts.Expression | undefined {
+    switch (statement.kind) {
+        case ts.SyntaxKind.ImportDeclaration:
+            return (statement as ts.ImportDeclaration).moduleSpecifier;
+        case ts.SyntaxKind.ImportEqualsDeclaration:
+            const ref = (statement as ts.ImportEqualsDeclaration).moduleReference;
```

Comment:
It didn't check ImportEqualsDeclaration before your change. That's not a problem, but should be mentioned in the changelog.

Reply:
It was right before because you might want named imports from a module in addition to importing it as a function.